src/v1/parser.rs

```diff
@@ -124,14 +124,99 @@ quick_error! {
 }
 
 /// Represents a type of the entry inside a signature file
-#[derive(Debug)]
+#[derive(Debug, PartialEq, Eq)]
 pub enum EntryKind<P: AsRef<Path>> {
     /// A directory
     Dir(P),
     /// A file or a symbolic link
     File(P),
 }
 
+impl<P: AsRef<Path>> EntryKind<P> {
+    /// Get path of the entry
+    pub fn path(&self) -> &Path {
+        use self::EntryKind::*;
+        match *self {
+            Dir(ref p) | File(ref p) => p.as_ref(),
+        }
+    }
+
+    /// Converts to EntryKind<&Path>
+    pub fn as_ref(&self) -> EntryKind<&Path> {
+        use self::EntryKind::*;
+        match *self {
+            Dir(ref p) => Dir(p.as_ref()),
+            File(ref p) => File(p.as_ref()),
+        }
+    }
+
+    /// Clones path and returns EntryKind<PathBuf>
+    pub fn cloned(&self) -> EntryKind<PathBuf> {
+        use self::EntryKind::*;
+        match *self {
+            Dir(ref p) => Dir(p.as_ref().to_path_buf()),
+            File(ref p) => File(p.as_ref().to_path_buf()),
+        }
+    }
+}
+
+impl<P> PartialOrd for EntryKind<P>
+    where P: AsRef<Path> + PartialEq + Eq
+{
+    fn partial_cmp(&self, other: &EntryKind<P>)
+        -> Option<Ordering>
+    {
+        Some(self.cmp(other))
+    }
+}
+
+impl<P> Ord for EntryKind<P>
+    where P: AsRef<Path> + PartialEq + Eq
+{
+    fn cmp(&self, other: &EntryKind<P>) -> Ordering {
+        use std::cmp::Ordering::*;
+        use self::EntryKind::*;
+
+        match *self {
```

Comment:
Yes, so basically I have misunderstood comparison rules. I'll take another look at it later, maybe I can find a better way to express this sort order. Or at least some comments might be added.